index.js

```diff
@@ -267,10 +282,16 @@ ShareDbMongo.prototype._writeSnapshot = function(collectionName, id, snapshot, o
         callback(null, true);
       });
     } else {
-      collection.replaceOne({_id: id, _v: doc._v - 1}, doc, function(err, result) {
-        if (err) return callback(err);
-        var succeeded = !!result.modifiedCount;
-        callback(null, succeeded);
+      request.query = {_id: id, _v: request.documentToWrite._v - 1};
+      self._middleware.trigger(MiddlewareHandler.Actions.beforeEdit, request, function(middlewareErr) {
```

Comment:
I wonder if we should generally avoid using `doc` in this library (particularly for public APIs), because of its meaning in `sharedb`. We could use `document` to refer to MongoDB documents?

Reply:
Agreed -since I did change `doc` in the `request` to `documentToWrite`

For simplicity's sake, what about just `beforeOverwrite`? That, coupled with the documentation should be clear enough.

Reply:
`beforeOverwrite` sounds good to me! It's sufficiently generic that we could still swap the implementation to do a field-by-field update in the future instead of a full-doc replace.